src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -1075,6 +1088,371 @@ describe('Transactions Controller (Unit)', () => {
+        creationTransactionBuilder()
+          .with('created', new Date(moduleTransaction.executionDate))
+          .build(),
+      );
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrl = `${safeConfigApiUrl}/api/v1/chains/${chainId}`;
+        const getAllTransactions = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}/all-transactions/`;
+        const getSafeUrl = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}`;
+        const getContractUrl = `${chainResponse.transactionService}/api/v1/contracts/`;
+        const getSafeCreationUrl = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}/creation/`;
+        if (url === getChainUrl) {
+          return Promise.resolve({ data: chainResponse });
+        }
+        if (url === getAllTransactions) {
+          return Promise.resolve({
+            data: allTransactionsResponse,
+          });
+        }
+        if (url === getSafeUrl) {
+          return Promise.resolve({
+            data: safe,
+          });
+        }
+        if (url === getSafeCreationUrl) {
+          return Promise.resolve({
+            data: creationTransactionResponse,
+          });
+        }
+        if (url.includes(getContractUrl)) {
+          return Promise.reject({
+            detail: 'Not found',
+          });
+        }
+        return Promise.reject(new Error(`Could not match ${url}`));
+      });
+
+      await request(app.getHttpServer())
+        .get(`/chains/${chainId}/safes/${safeAddress}/transactions/history/`)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.results).toHaveLength(3);
+          expect(body.results[2].transaction.timestamp).toEqual(
+            creationTransactionResponse.created.getTime(),
+          );
+          expect(body.results[2].transaction.txInfo.creator.value).toEqual(
+            creationTransactionResponse.creator,
+          );
+          expect(body.results[2].transaction.txInfo.factory.value).toEqual(
+            creationTransactionResponse.factoryAddress,
+          );
+          expect(
+            body.results[2].transaction.txInfo.implementation.value,
+          ).toEqual(creationTransactionResponse.masterCopy);
+          expect(body.results[2].transaction.txInfo.transactionHash).toEqual(
+            creationTransactionResponse.transactionHash,
+          );
+        });
+    });
+
+    it('Should keep the client pagination', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const chainResponse = chainBuilder().with('chainId', chainId).build();
+      const limit = 5;
+      const offset = 5;
+      const moduleTransaction = moduleTransactionTojson(
+        moduleTransactionBuilder().with('dataDecoded', null).build(),
+      );
+      const safe = safeBuilder().build();
+      const client_next_cursor = `cursor=limit%3D${limit}%26offset%3D10`;
```

Comment:
Nitpick: we should use camelCase to name variables I think 🙂

Reply:
yes I did it here 9bff7de84cb59f256094b48f1d8da232c1e29ac2, also for the other similar comments